src/Apps/Order/Components/ExpressCheckout/Util/mutationHandling.ts

```diff
@@ -0,0 +1,38 @@
+type OrderMutationSuccess<T> = Extract<
+  T,
+  { __typename: "OrderMutationSuccess" }
+>
+type OrderMutationError<T> = Extract<
+  T,
+  { __typename: "OrderMutationError"; mutationError: { message: string } }
+>
+
+const isOrderMutationSuccess = <T extends { __typename?: string }>(
+  data: T | null | undefined,
+): data is OrderMutationSuccess<T> => {
+  return data?.__typename === "OrderMutationSuccess"
+}
+
+const isOrderMutationError = <T extends { __typename?: string }>(
+  data: T | null | undefined,
+): data is OrderMutationError<T> => {
+  return data?.__typename === "OrderMutationError"
+}
+
+export const validateAndExtractOrderResponse = <
+  T extends { __typename?: string },
+>(
+  orderOrError?: T | null | undefined,
+): OrderMutationSuccess<T> => {
+  if (isOrderMutationSuccess(orderOrError)) {
+    return orderOrError
+  }
+
+  if (isOrderMutationError(orderOrError)) {
+    throw new Error(orderOrError.mutationError.message)
+  }
+
+  throw new Error(
+    `Unhandled orderOrError response type: ${JSON.stringify(orderOrError)}`,
+  )
+}
```

Comment:
proud of this, it allows taking an `orderOrError` response and turning it into a success or throwing (since the mutation is likely already in a `try`

Reply:
Nice, I was actually just working on a very similar mutation helper over in volt, since we're about to start writing a ton of mutations and don't want to deal with all of the verbosity of our mutation pattern

Reply:
I had to make a single mutation response union to be shared in metaphysics so hopefully this doesn't become problematic later. Also currently exchange throws known errors with a single `message` which is probably actually the exchange `code`. Might be able to improve on this as we proceed but that should make it a little easier to match/fall back on in clients when it is a known situation, like
```tsx
try {
  const result = updateOrder.submitMutation({})
  const data = validateAndExtractOrderResponse(result.updateOrder)
  
  resolveWith(data.order)
} catch (e) {
  handleKnownErrors(e) || doSomethingElse(e)
}
```